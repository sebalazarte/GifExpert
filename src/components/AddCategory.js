import React, { useState, Fragment } from 'react'
import PropTypes from 'prop-types'


export const AddCategory = ({ setCategorias, setLimite }) => { //se desestructuro props

    const [inputValue, setInputValue] = useState('');
    const [numberValue, setNumberValue] = useState(10);

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleNumberChange = (e) => {
        setNumberValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setCategorias(cats => [inputValue, ...cats,]);
            setInputValue('');
        }

        if (numberValue.length > 0) {
            setLimite(numberValue);
        }
    }

    const handleClean = (e) => {
        e.preventDefault();
        setCategorias([]);
    }

    return (
        <Fragment>
            <form onSubmit={handleSubmit}>
                <div className="row g-3">
                    <div className="col-md-8">
                        <div className="input-group mb-3">
                            <label className="input-group-text">Buscar</label>
                            <input
                                id="txtBuscar"
                                type="text"
                                value={inputValue}
                                onChange={handleInputChange}
                                className="form-control"
                                placeholder="ingrese para buscar..."
                                required />
                            <label className="input-group-text">Limite</label>
                            <input
                                id="txtCantidad"
                                type="number"
                                value={numberValue}
                                onChange={handleNumberChange}
                                className="form-control cantidad"
                            />
                            <button type="submit" className="btn btn-primary">
                                Buscar
                            </button>
                        </div>
                    </div>
                    <div className="col-md-4">
                        <button onClick={handleClean} className="btn btn-danger">
                            Limpiar
                        </button>   
                    </div>
                </div>
            </form>

        </Fragment>
    )
}

AddCategory.propTypes = {
    setCategorias: PropTypes.func.isRequired
}