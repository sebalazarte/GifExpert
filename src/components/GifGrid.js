import React, {Fragment } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';


export const GifGrid = ({ category, limit }) => {

    const {data: images, loading} = useFetchGifs(category, limit);

    return (
        <Fragment>
            <h3 className="animate__animated animate__faden">{category}</h3>
            
            {loading && <p className="animate__animated animate__flash">Cargando...</p>}

            <div className="card-grid">

                {
                    images.map(img =>
                        <GifGridItem
                            key={img.id}
                            {...img} />)
                }
            </div>
        </Fragment>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
    limit: PropTypes.number.isRequired
}