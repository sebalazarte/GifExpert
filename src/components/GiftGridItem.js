import React from 'react'

export const GiftGridItem = ({id, title, url}) => {
    // console.log(props);
    return (
        <div className="card animate__animated animate__bounceIn">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}
