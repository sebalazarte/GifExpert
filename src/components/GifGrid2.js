import React, { useState, useEffect, Fragment } from 'react'
import { getGifts } from '../helpers/getGifs';
import { GiftGridItem } from './GiftGridItem';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    useEffect(()=>{
        getGifts(category)
            .then( setImages );
    }, [category]);

    return (
        <Fragment>
            <h3>{category}</h3>
            <div className="card-grid">

                {
                    images.map(img =>
                        <GiftGridItem
                            key={img.id}
                            {...img} />)
                }
            </div>
        </Fragment>
    )
}
