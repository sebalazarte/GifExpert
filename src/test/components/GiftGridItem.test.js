import React from 'react';
import {shallow} from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GiftGridItem/>', () => {

    const url = 'https://media1.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=70a38d38casbobveplnboyjb4ahtzdhdxe3dfv9dztcr6q6d&rid=giphy.gif&ct=g';
    const title = 'Dragon Ball GIF by TOEI Animation UK';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('Debe mostrar el control correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe tener un parrafo con el texto ', () => {
        
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('Debe tener una imagen con url y alt', () => {

        const img = wrapper.find('img');
        const {src, alt} = img.props();

        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe tener una animate__fadeIn', () => {

        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className).toContain('animate__fadeIn');
    });

});