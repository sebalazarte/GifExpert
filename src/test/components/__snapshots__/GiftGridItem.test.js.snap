// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftGridItem/> Debe mostrar el control correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Dragon Ball GIF by TOEI Animation UK"
    src="https://media1.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=70a38d38casbobveplnboyjb4ahtzdhdxe3dfv9dztcr6q6d&rid=giphy.gif&ct=g"
  />
  <p>
    Dragon Ball GIF by TOEI Animation UK
  </p>
</div>
`;
